packages/sql-pglite/src/PgLiteClient.ts

```diff
@@ -0,0 +1,343 @@
+        })
+      }
+
+      execute(
+        sql: string,
+        params: ReadonlyArray<Primitive>,
+        transformRows?: (<A extends object>(row: ReadonlyArray<A>) => ReadonlyArray<A>) | undefined,
+        unprepared?: boolean
+      ) {
+        return transformRows
+          ? Effect.map(
+            this.run(
+              unprepared ? this.pg.exec(sql, params as any) : this.pg.query(sql, params as any)
+            ),
+            transformRows
+          )
+          : unprepared
+          ? this.run(this.pg.exec(sql, params as any))
+          : this.run(this.pg.query(sql, params as any))
+      }
+      executeRaw(sql: string, params: ReadonlyArray<Primitive>) {
+        return this.run(this.pg.exec(sql, params as any))
+      }
+      executeWithoutTransform(sql: string, params: ReadonlyArray<Primitive>) {
+        return this.run(this.pg.query(sql, params as any))
+      }
+      executeValues(sql: string, params: ReadonlyArray<Primitive>) {
+        // PGlite doesn't have a values() method like postgres.js
+        // We'll just return the regular query results
+        return this.execute(sql, params, (r) => Object.values(r))
+      }
+      executeUnprepared(
+        sql: string,
+        params: ReadonlyArray<Primitive>,
+        transformRows: (<A extends object>(row: ReadonlyArray<A>) => ReadonlyArray<A>) | undefined
+      ) {
+        return this.execute(sql, params, transformRows, true)
+      }
+      executeStream(
+        sql: string,
+        params: ReadonlyArray<Primitive>,
+        transformRows: (<A extends object>(row: ReadonlyArray<A>) => ReadonlyArray<A>) | undefined
+      ) {
+        // PGlite doesn't have a cursor method like postgres.js
+        // We'll fetch all results at once and convert to a stream
+        return Stream.fromEffect(
+          Effect.map(this.run(this.pg.query(sql, params as any)), (rows) => {
+            const result = transformRows ? transformRows(rows) : rows
+            return result
+          })
+        ).pipe(Stream.flatMap(Stream.fromIterable))
+      }
+    }
+
+    return Object.assign(
+      yield* Client.make({
+        acquirer: Effect.succeed(new ConnectionImpl(client)),
+        compiler,
+        spanAttributes: [
+          ...(options.spanAttributes ? Object.entries(options.spanAttributes) : []),
+          [OtelSemConv.ATTR_DB_SYSTEM_NAME, OtelSemConv.DB_SYSTEM_NAME_VALUE_POSTGRESQL],
+          [OtelSemConv.ATTR_DB_NAMESPACE, options.database ?? options.username ?? "postgres"]
+        ],
+        transformRows
+      }),
+      {
+        [TypeId]: TypeId as TypeId,
+        config: {
+          ...options
+        },
+        client,
+        json: (_: unknown) => PgliteJson([_]),
+        array: (_: ReadonlyArray<Primitive>) => PgliteArray([_]),
+        extensions: options.extensions ? (client as any) : ({} as any),
+        listen: (channel: string) =>
+          Stream.asyncPush<string, SqlError>((emit) =>
+            Effect.tryPromise({
```

Comment:
You would use Effect.acquireRelease here. I think this would immediately unsubscribe.